src/action-sheets/index.js

```diff
@@ -214,6 +215,49 @@ const muteStream = async ({ auth, streamId, subscriptions }) => {
 muteStream.title = 'Mute stream';
 muteStream.errorMessage = 'Failed to mute stream';
 
+const hashReplacements: Map<string, string> = new Map([
+  ['%', '.'],
+  ['(', '.28'],
+  [')', '.29'],
+  ['.', '.2E'],
+]);
+
+// we trying to create uri like web app.
+// see https://github.com/zulip/zulip/blob/HEAD/static/js/hash_util.js
+const encodeHashComponent = (str: string) =>
+  encodeURIComponent(str).replace(/[%().]/g, matched => hashReplacements.get(matched) ?? '');
+
+const getStreamDetail = (streamId, streams) => {
+  const stream = streams.get(streamId);
+  invariant(stream !== undefined, 'Stream with provided streamId not found.');
+
+  // The name part of the URL doesn't really matter, so we try to
+  // make it pretty.
+  const streamName = stream.name.replace(' ', '-');
```

Comment:
Looking closer at the relevant web-app code, there's something missing here.

Instead of throwing an error when the stream isn't found, it falls back to the name "unknown". I'm not sure that's the ideal behavior (for one thing, that string doesn't get translated), but we should default to copying the web app, and then mark any interesting changes we make.

Probably simplest for this PR to just copy the web app.